ddtrace/contrib/boto/patch.py

```diff
@@ -0,0 +1,92 @@
+import ddtrace
+
+import boto.connection
+import wrapt
+import pkg_resources
+from distutils.version import LooseVersion
+
+from ...ext import AppTypes
+from ...ext import http
+
+# Original boto client class
+_Boto_client = boto.connection.AWSQueryConnection
+
+DEFAULT_SERVICE = "aws"
+SPAN_TYPE = "boto"
+
+
+def patch():
+
+    """ AWSQueryConnection and AWSAuthConnection are two different classes called by
+    different services for connection. For exemple EC2 uses AWSQueryConnection and
+    S3 uses AWSAuthConnection
+    """
+    # Checking on the version for compatibility before patching
+    if LooseVersion(pkg_resources.get_distribution("boto").version) >= LooseVersion("2.29.0"):
+        wrapt.wrap_function_wrapper('boto.connection', 'AWSQueryConnection.make_request', patched_query_request)
+        wrapt.wrap_function_wrapper('boto.connection', 'AWSAuthConnection.make_request', patched_auth_request)
+
+# ec2, sqs, kinesis
+def patched_query_request(original_func, instance, args, kwargs):
+    tracer = getattr(instance, 'datadog_tracer', ddtrace.tracer)
+    tracer.set_service_info(service=DEFAULT_SERVICE, app=SPAN_TYPE, app_type=AppTypes.web)
+
+    if not tracer.enabled:
+        return original_func(*args, **kwargs)
+
+    with tracer.trace("aws.api.request") as span:
+        operation_name, _, _, _ = args
+        span.set_tag('aws.operation', operation_name)
+        # Obtaining region name
+        region = getattr(instance, "region", ":")
+        span.set_tag('aws.region', get_region_name(region))
+
+        # Obtaining endpoint name and region name
+        host = getattr(instance, "host", "unknown.unknown..").split('.')
+        endpoint_name = "unknown"
+        if len(host) == 4:
+            endpoint_name = host[0]
+        span.set_tag('aws.endpoint', endpoint_name)
+
+        # Original func returns a boto.connection.HPPResponse object
+        result = original_func(*args, **kwargs)
+        span.set_tag(http.STATUS_CODE, getattr(result, "status", 500))
```

Comment:
let's not have it default to 500. This is a meaningful error code that influences downstream stats collection and error reporting/alerting so we do not want to set it just because there's no better candidate. Instead it's fine to set the STATUS_CODE meta only when `result.status` is available